### ISSUE TITLE
New files are added to csproj when EnableDefaultItems is True

### ISSUE THREAD

**Issue by grandelbert** [dotnet/project-system#7175]
**Visual Studio Version**:
16.9.4

**Summary**:
When adding a new file to a .NET core/standard project via the Solution Explorer, it is automatically added to the .csproj file, even when `EnableDefaultItems` is explicitly set to `True`.

e.g. I create the file `XMLFile1.xml`, then this is added to the .csproj
``` xml
  <ItemGroup>
    <Content Include="XMLFile1.xml" />
  </ItemGroup>
```

This causes the `NETSDK1022 Duplicate 'Compile' items were included.` error.

@AdamCaviness has done further testing (reinstalling, cleaning registry etc.) on this and [documented it here](https://developercommunity.visualstudio.com/t/compile-elements-added-to-project-file-when-enable/1356490). I have also [added a description](https://developercommunity.visualstudio.com/t/compile-elements-added-to-project-file-when-enable/1356490#T-N1412147) of a weird, probably related problem when moving files from one directory to another via the Solution Explorer. Should I create another issue for that?

This problem has also been reported by some other people, but never resolved.
https://docs.microsoft.com/en-us/answers/questions/309774/netsdk1022.html
https://stackoverflow.com/questions/65557809/prevent-vs-from-adding-explicit-compile-tags-to-csproj
https://youtrack.jetbrains.com/issue/RSRP-483917 & https://developercommunity.visualstudio.com/t/Renaming-a-file-removes-all-files-from-s/1389846

There was also a report where the opposite problem occurred (files not added when EnableDefaultItems was False) #3181.
No idea if this is related or this is even a Project System problem.

**Steps to Reproduce**:
Looks like it is not reproducable on most installations.
All I have to do is create a new file for a project.

**Expected Behavior**:
.csproj file should not be modified when `EnableDefaultItems` (or `EnableDefaultCompileItems` etc. for that matter) is enabled either explicitly in the .csproj or by globs.

**Actual Behavior**:
No matter whether it is enabled by globs (.NET 5 defaults) or in the .csproj file, new entries in .csproj are made.

**User Impact**:
It is rather inconvenient, as you have to create new files manually or with another editor or edit the csproj each time before you can build.

**Comment by jjmew** [comment]
@drewnoakes Did you investigate something similar before?

**Comment by drewnoakes** [comment]
@jjmew it doesn't sound familiar. I suspect this would be CPS.

**Comment by jjmew** [comment]
We have move this on an internal database where CPS tracks there issues. 

https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1339843

**Comment by lifengl** [comment]
It looks like this issue has been reported by multiple user. However, it can only be reproduced by those who ran into it, which makes this very hard to investigate.

Unfortunately, this code path is complex, and can be affected by extensions, imported files in the projects, item template and maybe something else. It is hard to identify the source of the problem.

Now, if anyone still runs into this issue can help to collect some information, we hopefully can make some progress in the future, otherwise, i don't see it is actionable.

First, I would suggest those who are setting it to turn on project system diagnostic feature by setting environment variable CPS_DiagnosticRuntime to "1" to the system, before starting Visual studio.  After that, select the project which is affected by this, and in the toolbar of the solution explorer, right after the "show all files" button, there is a new "Toggle CPS diagnostic features" button, and turn it on. A capabilities tree will show under the project node, expand it and capture what capabilities are turned on in the project.  For example, if "UseFileGlobs" is turned off, explicit item will be added to the project just like traditional projects. Some other capabilities might turn on behaviors which unexpected to be here, so that is the first information we want to gather. 

This will also show project logging data inside a new "project" tab in the output window. Currently it doesn't log much information to help our investigation, which we plan to add something in 17.1.  However, if no one still runs into this anymore (one feedback was closed after 16.10 updates), We will close this issue.
